node/node.go

```diff
@@ -1106,10 +1109,17 @@ func (app *App) launchStandalone(ctx context.Context) error {
 	cfg := server.DefaultConfig()
 	cfg.PoetDir = filepath.Join(app.Config.DataDir(), "poet")
 
-	parsed, err := url.Parse(app.Config.PoETServers[0])
+	parsed, err := url.Parse(app.Config.PoetServers[0].Address)
 	if err != nil {
 		return err
 	}
+	pubkey, privkey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return err
+	}
+	app.Config.PoetServers[0].Pubkey = types.NewBase64Enc(pubkey)
+	os.Setenv("POET_PRIVATE_KEY", base64.StdEncoding.EncodeToString(privkey))
```

Comment:
Probably out of scope for this PR, but this is backwards. The PoET should generate the keys and pass the public key to the node, not the node passing the private key to poet.

Reply:
Fixed